core/src/main/java/com/linecorp/armeria/common/metric/NettyMetrics.java

```diff
@@ -0,0 +1,133 @@
+/*
+ *  Copyright 2023 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package com.linecorp.armeria.common.metric;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import com.linecorp.armeria.common.util.Ticker;
+import com.linecorp.armeria.internal.common.metric.MicrometerUtil;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.SingleThreadEventLoop;
+import io.netty.util.concurrent.EventExecutor;
+
+/**
+ * A {@link MeterBinder} to observe Netty's runtime. The following stats are currently exported per
+ * registered {@link MeterIdPrefix}.
+ *
+ * <p>
+ *  - "event.loops.num.workers" (gauge) - the total number of Netty's event loops
+ *  - "event.loops.pending.tasks" (gauge) - the total number of IO tasks waiting to be run on event loops*
+ * </p>
+ */
+public final class NettyMetrics implements MeterBinder {
```

Comment:
- Hide from `public`?
- Should we copy the Javadoc to `MoreMetrics.nettyMetrics()` since users can see Javadoc of package-private in the [API documentation](https://javadoc.io/doc/com.linecorp.armeria/armeria-javadoc/latest/index.html)?
- `NettyMetrics` sounds like it exposes all Netty-related metrics. We will add #2633 which may not be added to this class. Because users should use `MoreMetrics.nettyMetrics()` to monitor their custom `EventLoopGroups`. What do you think of renaming  `NettyMetrics` to `EventLoopGroupMetrics` or something similar?
```suggestion
 * <ul>
 *  <li>"event.loops.num.workers" (gauge) - the total number of Netty's event loops</li>
 *  <li>"event.loops.pending.tasks" (gauge) - the total number of IO tasks waiting to be run on event loops</li>
 * </ul>
 */
final class NettyMetrics implements MeterBinder {
```

Reply:
> What do you think of renaming NettyMetrics to EventLoopGroupMetrics or something similar?

Hmm, I was actually thinking the allocator metrics would go here too but I can see the argument for better granularity. I will rename.